tools/src/launcher/Main.hs

```diff
@@ -178,16 +183,92 @@ reportErrorDefault filename contents = do
     createDirectoryIfMissing True logDir
     writeFile (logDir </> filename) contents
 
+-- * Environment variable parsing and substitution for the launcher configuration file,
+--   typically launcher-config.yaml.
+
+-- | Represent an element of a parse of a string containing environment variable
+--   references: either a plain text piece, or a variable reference.
+data Chunk
+  = EnvVar Text
+  | Plain  Text
+  deriving (Show)
+
+-- | A parse is a sequence of variable references & plain text pieces.
+parseEnvrefs :: Text -> P.Result [Chunk]
+parseEnvrefs text = P.parseString pEnvrefs mempty (toString text)
+  where
+    pEnvrefs :: (Monad p, P.TokenParsing p) => p [Chunk]
+    pEnvrefs = P.some $ P.choice [pPassiveText, pEnvvarRef]
+
+    pEnvvarRef, pPassiveText :: (Monad p, P.TokenParsing p) => p Chunk
+    -- TODO: figure out a less ugly OS conditionalisation method
```

Comment:
Would https://hackage.haskell.org/package/Cabal-1.22.2.0/docs/Distribution-System.html be appropriate?

Reply:
Or simply the base package's `System.Info.os`. That can usually avoid needing cpp and doesn't add deps.

Reply:
Thank you!